torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1481,6 +1481,15 @@ def __init__(self, name, *, lhs_make_tensor_kwargs=None, rhs_make_tensor_kwargs=
             rhs_make_tensor_kwargs = {}
         self.rhs_make_tensor_kwargs = rhs_make_tensor_kwargs
 
+        if result_type is None:
+            result_type = {}
+        self._result_type = result_type
+
+    def result_type(self, lhs, rhs):
+        lhs_dtype = lhs.dtype if isinstance(lhs, torch.Tensor) else type(lhs)
+        rhs_dtype = rhs.dtype if isinstance(rhs, torch.Tensor) else type(rhs)
```

Comment:
Although this is not needed here, this also enables the case of overwriting the default behavior for scalar inputs, e.g. `torch.foo(torch.tensor(..., dtype=torch.int32), 1.0)` can be overwritten with `result_type={(torch.int32, float): ...}`